tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -129,15 +122,24 @@ bool IsHLSLNumericUserDefinedType(clang::QualType type) {
   return false;
 }
 
+bool IsHLSLBuiltinRayAttributeStruct(clang::QualType QT) {
+  QT = QT.getCanonicalType();
+  const clang::Type *Ty = QT.getTypePtr();
+  if (const RecordType *RT = dyn_cast<RecordType>(Ty)) {
+    const RecordDecl *RD = RT->getDecl();
+    if (RD->getName() == "RaytracingAccelerationStructure" ||
+        RD->getName() == "BuiltInTriangleIntersectionAttributes")
```

Comment:
These two shouldn't be treated the same.  The first (`RaytracingAccelerationStructure`) is an opaque resource type, while the second (`BuiltInTriangleIntersectionAttributes`) is a pre-defined structure you can use just like any normal numeric structure, in particular, you may use this for the attribute struct "user-defined-type" in a `[shader("closesthit")]` or other shader that takes hit attributes.

Reply:
`RaytracingAccelerationStructure` should not be legal for use as numeric struct, so that's the one that will be a problem here.